TFNetwork.py

```diff
@@ -383,18 +383,26 @@ def construct_layer(self, net_dict, name, get_layer=None, add_layer=None, check_
     if name in self._constructing_layers:
       raise NetworkConstructionDependencyLoopException(
         layer_name=name, constructing_layers=self._constructing_layers, net_dict=net_dict, network=self)
+    if not get_layer:
+      def get_layer(src_name):
+        return self.construct_layer(net_dict=net_dict, name=src_name)  # set get_layer to wrap construct_layer
     if name not in net_dict:
       if name == "data":
         layer_desc = {"class": "source", "from": []}
       elif name.startswith("data:"):
         layer_desc = {"class": "source", "data_key": name[len("data:"):], "from": []}
+      elif '/' in name:
+        # it may be a hierarchical path to a sub-layer, which should have been found by get_layer()
+        # but maybe it's not constructed yet, so try constructing the root layer
+        get_layer(name.split('/')[0])
+        assert name.split('/')[0] in self.layers, "Root %r layer was not constructed! " \
```

Comment:
Small detail: Instead of `\` for splitting this into two lines, I prefer to use brackets, like `assert ...., ("Root ... ! "` newline `"Unable to ..." % (...))`.
(Despite personal taste, there is also one small advantage of this: When an exception would occur here, the code line in the stack trace would be the last line of the statement, i.e. in your case the `"Unable to ..." % ..`. When showing that statement alone in a debug stack trace, you might miss that this is part of another statement. E.g. better_exchook looks if the stack frame code line contains any non-matching `)`, and only then it would show also the preceding code line.)